stdlib/list.mli

```diff
@@ -128,6 +128,13 @@ val flatten : 'a list list -> 'a list
    (length of the argument + length of the longest sub-list).
  *)
 
+val intersperse : 'a -> 'a list -> 'a list
+(** [intersperse sep l] is the list formed by inserting [sep] between adjacent
```

Comment:
Why not go further and allow the separator to be a `'a list`?

Reply:
The `'a -> 'a list -> 'a list` form has a more descriptive type and I think more intuitively matches the name "intersperse".

Anecdotally, I've never found I needed the more general form either. YMMV.

Reply:
Personally I'm not sure I had that much of a need for `intersperse` (but I don't mind). I rather often missed the signature of `String.concat` for `List` (which would be `'a list -> 'a list list -> 'a list`).

Reply:
@dbuenzli: Interesting. I've never found a need for that one, but I know Haskell developers like it, and they call it `intercalate`. I'm happy to add that along with this PR if it's deemed worthy; it's in some sense the natural sibling to `intersperse`.